Core/clim-basic/ports.lisp

```diff
@@ -168,12 +169,96 @@
+                    (t (setf sheet-a (sheet-parent sheet-a))))))
+         (get-pointer-event-sheet (sheet event)
+           (get-pointer-position (sheet event)
+             (loop
+                ;; only not mirrored child?
+                for child = (child-containing-position sheet x y)
+                do
+                  (when (null child)
+                    (return sheet))
+                  (multiple-value-setq (x y)
+                    (untransform-position (sheet-transformation child) x y))
+                  (setf sheet child))))
+         (distribute-enter-events (sheet-b sheet-t event)
+           (dolist (s
+                     (do ((s sheet-b (sheet-parent s))
+                          (lis nil))
+                         ((or (null s) (graftp s) (eq s sheet-t)) lis)
+                       (push s lis)))
+             (let ((new-event (shallow-copy-object event)))
+               ;; should we change also `x' and `y'?
+               (setf (slot-value new-event 'sheet) s)
+               (change-class new-event 'pointer-enter-event)
+               (dispatch-event s new-event))))
+         (distribute-exit-events (sheet-b sheet-t event)
+           (when (and sheet-t sheet-b)
+             (do ((s sheet-b (sheet-parent s)))
+                 ((or (null s) (graftp s) (eq s sheet-t)))
+               (let ((new-event (shallow-copy-object event)))
+                 ;; should we change also `x' and `y'?
+                 (setf (slot-value new-event 'sheet) s)
+                 (change-class new-event 'pointer-exit-event)
+                 (dispatch-event s new-event))))))
+    (let* ((pointer-sheet (get-pointer-event-sheet (event-sheet event) event))
+           (old-pointer-sheet (or (port-pointer-sheet port) (event-sheet event)))
+           (common-sheet (sheet-common-ancestor old-pointer-sheet pointer-sheet)))
+      (distribute-exit-events old-pointer-sheet common-sheet event)
+      (distribute-enter-events pointer-sheet common-sheet event)
+      (setf (port-pointer-sheet port) pointer-sheet)
+      ;; set the pointer cursor
+      (when pointer-sheet
+        (let ((pointer-cursor (sheet-pointer-cursor pointer-sheet)))
+	  (unless (eql (port-lookup-current-pointer-cursor port (event-sheet event))
+		       pointer-cursor)
+	    (set-sheet-pointer-cursor port (event-sheet event) pointer-cursor))))))
+  ;; Do not send the event twice.
+  (unless (or (typep event 'pointer-enter-event)
+	      (typep event 'pointer-exit-event))
+    (call-next-method)))
+
+;;; This method is necessary to distribute events for sheets which
+;;; does not have a mirror. In this case we find the innermost sheet
+;;; at event's position and dispatch event's shallow copy to this
+;;; sheet. -- jd 2019-08-05
+(defmethod distribute-event ((port basic-port) (event pointer-event))
+  (when-let ((sheet (port-pointer-sheet port)))
+    (if (eq sheet (event-sheet event))
+        (dispatch-event sheet event)
```

Comment:
Can we `(call-next-method)` here? Subclasses might define their own `distribute-event ((port my-port) event` method.

Reply:
fixed